src/import/FSHImporter.ts

```diff
@@ -0,0 +1,393 @@
+    }
+    console.log('Unsupported rule: ', ctx.getText());
+    return [];
+  }
+
+  visitPath(ctx: pc.PathContext): string {
+    return ctx.SEQUENCE().getText();
+  }
+
+  visitPaths(ctx: pc.PathsContext): string[] {
+    return ctx
+      .COMMA_DELIMITED_SEQUENCES()
+      .getText()
+      .split(/,\s+/);
+  }
+
+  visitCardRule(ctx: pc.CardRuleContext): (CardRule | FlagRule)[] {
+    const rules: (CardRule | FlagRule)[] = [];
+
+    const cardRule = new CardRule(this.visitPath(ctx.path()));
+    const card = this.parseCard(ctx.CARD().getText());
+    cardRule.min = card.min;
+    cardRule.max = card.max;
+    rules.push(cardRule);
+
+    if (ctx.flag() && ctx.flag().length > 0) {
+      const flagRule = new FlagRule(cardRule.path);
+      this.parseFlags(flagRule, ctx.flag());
+      rules.push(flagRule);
+    }
+    return rules;
+  }
+
+  private parseCard(card: string): { min: number; max: string } {
+    const parts = card.split('..', 2);
+    return {
+      min: parseInt(parts[0]),
+      max: parts[1]
+    };
+  }
+
+  visitFlagRule(ctx: pc.FlagRuleContext): FlagRule[] {
+    let paths: string[];
+    if (ctx.path()) {
+      paths = [this.visitPath(ctx.path())];
+    } else if (ctx.paths()) {
+      paths = this.visitPaths(ctx.paths());
+    }
+
+    return paths.map(path => {
+      const flagRule = new FlagRule(path);
+      this.parseFlags(flagRule, ctx.flag());
+      return flagRule;
+    });
+  }
+
+  private parseFlags(flagRule: FlagRule, flagContext: pc.FlagContext[]): void {
+    const flags = flagContext.map(f => this.visitFlag(f));
+    flagRule.mustSupport = flags.includes(Flag.MustSupport);
+    flagRule.summary = flags.includes(Flag.Summary);
+    flagRule.modifier = flags.includes(Flag.Modifier);
+  }
+
+  visitFlag(ctx: pc.FlagContext): Flag {
+    if (ctx.KW_MS()) {
+      return Flag.MustSupport;
+    } else if (ctx.KW_SU()) {
+      return Flag.Summary;
+    } else if (ctx.KW_MOD()) {
+      return Flag.Modifier;
+    }
+    return Flag.Unknown;
+  }
+
+  visitValueSetRule(ctx: pc.ValueSetRuleContext): ValueSetRule {
+    const vsRule = new ValueSetRule(this.visitPath(ctx.path()));
+    vsRule.valueSet = this.aliasAwareValue(ctx.SEQUENCE().getText());
+    vsRule.strength = ctx.strength() ? this.visitStrength(ctx.strength()) : 'required';
+    return vsRule;
+  }
+
+  visitStrength(ctx: pc.StrengthContext): string {
+    if (ctx.KW_EXAMPLE()) {
+      return 'example';
+    } else if (ctx.KW_PREFERRED()) {
+      return 'preferred';
+    } else if (ctx.KW_EXTENSIBLE()) {
+      return 'extensible';
+    }
+    return 'required';
+  }
+
+  visitFixedValueRule(ctx: pc.FixedValueRuleContext): FixedValueRule {
+    const fixedValueRule = new FixedValueRule(this.visitPath(ctx.path()));
+    fixedValueRule.fixedValue = this.visitValue(ctx.value());
+    return fixedValueRule;
+  }
+
+  visitValue(ctx: pc.ValueContext): boolean | number | string | Code | Quantity | Ratio {
```

Comment:
Could it possible be a good idea to define a `FixedValueRuleType` so we don't have to re-write this union in two different places. @kjmahalingam brought this up to me when I was making the FixedValueRule class but it got lost in the shuffle.

Reply:
I think this is a good idea. I was wondering if it would be useful to make such a type in that previous PR in case we ever needed to use the same type set elsewhere at some point; and it looks like we did!

Reply:
Yeah, as I was writing that I thought it seemed kind of repetitive.  Thanks for the encouragement to go ahead and `type` it.